File: examples/basic-vars/test/integration/java/SayHelloStepTest.java
import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
import org.jenkinsci.plugins.workflow.job.WorkflowJob;
import org.jenkinsci.plugins.workflow.job.WorkflowRun;
import org.junit.Rule;
import org.junit.Test;
import org.jvnet.hudson.test.JenkinsRule;

public class SayHelloStepTest {

    @Rule
    public JenkinsRule jenkins = new JenkinsRule();

    @Test
    public void defaultGreeting() throws Exception {
        WorkflowJob job = jenkins.createProject(WorkflowJob.class);
        job.setDefinition(new CpsFlowDefinition("sayHello()", true));
        WorkflowRun run = jenkins.buildAndAssertSuccess(job);
        jenkins.assertLogContains("Hello, world!", run);
    }

    @Test
    public void namedGreeting() throws Exception {
        WorkflowJob job = jenkins.createProject(WorkflowJob.class);
        job.setDefinition(new CpsFlowDefinition("sayHello('Jenkins')", true));
        WorkflowRun run = jenkins.buildAndAssertSuccess(job);
        jenkins.assertLogContains("Hello, Jenkins!", run);
    }
}

File: examples/kotest/test/integration/kotlin/com/example/SayHelloStepTest.kt
package com.example

import io.kotest.matchers.string.shouldContain
import testsupport.jenkins.testharness.*
import testsupport.kotest.JenkinsFunSpec

class SayHelloStepTest : JenkinsFunSpec({
    test("default greeting") {
        jenkins { j ->
            val job = j.createWorkflowJob(definition = "sayHello()")
            val run = j.buildAndAssertSuccess(job)
            run.log shouldContain "Hello, world!"
        }
    }

    test("named greeting") {
        jenkins { j ->
            val job = j.createWorkflowJob(definition = "sayHello('Jenkins')")
            val run = j.buildAndAssertSuccess(job)
            run.log shouldContain "Hello, Jenkins!"
        }
    }
})
